libraries/protocol/include/golos/protocol/worker_operations.hpp

```diff
@@ -95,6 +95,26 @@ namespace golos { namespace protocol {
         }
     };
 
+    struct worker_assign_operation : public base_operation {
+        account_name_type techspec_author;
+        std::string techspec_permlink;
+        account_name_type worker;
+        bool assigned = true;
+        bool i_am_worker = false;
```

Comment:
When you have 2 bool fields -  you have 4 states: `false, false`, `true, false`, `false, true`, `true, true`
And you should check all this cases - it will increase complexity of the code.

Refactor to one field: `assigner` - where then assigner may be the `author` or the `worker` 
It will simplify code.

Current realization has too many possible cases for errors.